BCStudentSoftwareDevTeam/celts
# Created Volunteer Indicate Interest Section

This PR fixes issue #14 


**Summary:**

Created the following files:
_volunteerIndicateInterest.html_
_volunteerIndicateInterest.py_
_volunteerIndicateInterest.js_

In the _volunteerIndicateInterest.html_ we created the structure for our section. We added a head tag with the section title. Then, we added div tags looping through the programs' database listing all the programs. We then added checkboxes for every program. We also created a conditional that check the database for previously indicated interest and set the starting status of the checkboxes. We added another div tag to contain the flasher class. Finally, we created a small tag that contains the anchor tag, which leads to the CELTS program web page.

In the _volunteerIndicateInterest.js_, we created a function that onclick will be called to send the program_id and user_id to the controller in an Ajax call. To do this, we created a conditional that will check if a user is checking for interest or unchecking to remove interest. We then have the conditional select the appropriate route for deleting or adding interest. We have second conditional that will display a flasher if the Json is returned as Sucess.  

In the _volunteerIndicateInterest.py_, we created our imports and created three functions. The first function queries the database to pull in the appropriate data to render the template. And then in the second function, we have peewee create a row when the user indicates interest. In the third function, we have peewee delete a row when a user removes interest. 

**To Test:**
You can visit this URL: `http://<YOURIP>/volunteerIndicateInterest` you will see our basic section appear if you click on any options you can then check phpmyadmin and you will see the interest table update and if you refresh the page you will see your check box stays there. When you click you will also get a message that says your interests have been updated. If you uncheck a box you will see in php my admin the interest tabel has removed the row and if you refresh the page again you will see the program will stay unchecked. Below the table you will also see our link to the CELTS berea programs page.
